core/src/main/scala/sttp/client/testing/SttpBackendStub.scala

```diff
@@ -211,3 +282,100 @@ object SttpBackendStub {
     }
   }
 }
+
+/**
+  * Websockets use can differ, for akka backend crucial part is the handler.
+  * For other backends
+  */
+private[testing] sealed trait WhenOpenWebsocket[F[_], +WS_HANDLER[_]]
+private[testing] case class UseHandler[F[_], WS_RESULT, WS_HANDLER[_]](
+    headers: Headers,
+    useHandler: Any => WS_RESULT
+) extends WhenOpenWebsocket[F, WS_HANDLER]
+
+private[testing] case class ReturnWebsocketResponse[F[_], WS_RESULT](headers: Headers, response: F[WS_RESULT])
+    extends WhenOpenWebsocket[F, NothingT]
+
+/**
+  * Simple stub for web sockets that uses queue of events for receive.
+  */
+class WebSocketStub[F[_], S](
```

Comment:
I think to cover all use-cases we should also expose a method to enqueue events to be sent. This would be needed e.g. to simulate a websocket which periodically sends out a message.

Also, maybe we could make this thread-safe? Even in a "primitive" way, using an `AtomicReference` or maybe even a synchronized method.

Reply:
I retract the message-enqueueing method, this would require `receive` to get suspended whil waiting for new messages, and could cause tests to hang. That's I think outside the scope of a stub :)

But still would be nice to have things thread-safe :)